### PR TITLE
bring over etcd changes from modelmesh-serving

### PR DESCRIPTION
**Pull request by VedantMahabaleshwarkar**
## Description : 

This PR, along with [this](https://github.com/red-hat-data-services/odh-deployer/pull/309) and [this](https://github.com/red-hat-data-services/modelmesh-serving/pull/11) are meant to be tested together. Together, they address [RHODS-7487](https://issues.redhat.com/browse/RHODS-7487). etcd secret creation is moved from odh deployer to etcd initContainer. 

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-7487
- [x] The Jira story is acked
- [x] An entry has been added to the latest build document in [Build Announcements Folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL).
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious)

## Testing Instructions
- Install `quay.io/vedantm/rhods-operator-live-catalog:1.24.0-etcd`  
- Look at odh-deployer logs
  - search logs for `secret/etcd-passwords` and `secret/model-serving-etcd created`. result should be empty
  - in `redhat-ods-applications` look for the etcd pod
  - etcd Pod should have an initContainer with logs for 2 secrets created

### CODE REVIEW

**Comment by heyselbi** [comment]
tested and worked as expected

File: model-mesh/dependencies/quickstart.yaml [hunk 1]
```diff
@@ -47,6 +47,35 @@ spec:
           configMap:
             name: etcd-scripts
             defaultMode: 0554
+      initContainers: 
+        - name: etcd-secret-creator
+          image: quay.io/openshift/origin-cli
```
Comment (anishasthana) [inline model-mesh/dependencies/quickstart.yaml:52]: We need an image from registry.redhat, not quay.

**Review by anishasthana — CHANGES_REQUESTED** [review]

File: model-mesh/dependencies/quickstart.yaml (hunk 1 above)
Comment (VedantMahabaleshwarkar) [reply model-mesh/dependencies/quickstart.yaml]: updated image

**Review by anishasthana — APPROVED** [review]
/lgtm

**Comment by anishasthana** [comment]
image

**Pull request merged**